grails-app/views/stockMovement/show.gsp

```diff
@@ -75,78 +77,119 @@
                                 onclick="return confirm('${warehouse.message(code: 'default.button.delete.confirm.message', default: 'Are you sure?')}');"
-                                disabledMessage="You do not have minimum required role to delete stock movement"
+                                disabledMessage="You do not have minimum required role to delete stock request"
                         >
                             <img src="${resource(dir: 'images/icons/silk', file: 'delete.png')}" />&nbsp;
                             <warehouse:message code="default.button.delete.label" />
                         </g:link>
-                    </g:else>
-                </g:if>
-                <g:if test="${isPending && (isSameOrigin || isSameDestination || !originIsDepot) && stockMovement?.electronicType}">
-                    <g:link controller="stockRequest" action="remove" id="${stockMovement.id}" params="[show:true]" class="button"
-                            onclick="return confirm('${warehouse.message(code: 'default.button.delete.confirm.message', default: 'Are you sure?')}');"
-                            disabledMessage="You do not have minimum required role to delete stock request"
+                    </g:if>
+
+                <g:isSuperuser>
+                    <a href="javascript:void(0);" class="button btn-show-dialog"
+                        data-height="600" data-width="1000"
+                       data-title="${warehouse.message(code:'default.button.synchronize.label', default: 'Synchronize')}"
+                       data-url="${request.contextPath}/stockMovement/synchronizeDialog/${stockMovement?.id}">
+                        <img src="${resource(dir: 'images/icons/silk', file: 'arrow_join.png')}" />&nbsp;
+                        <warehouse:message code="default.button.synchronize.label" default="Synchronize"/>
+                    </a>
+                </g:isSuperuser>
+            </g:if>
+            <g:if test="${userHasRequestApproverRole}">
+                <g:supports location="${stockMovement.origin?.id}" activityCode="${ActivityCode.APPROVE_REQUEST}">
+                    <g:link
+                            class="button"
+                            controller="stockRequest"
+                            action="updateStatus"
+                            id="${stockMovement.id}"
+                            params="[status: StockMovementStatusCode.APPROVED]"
+                            disabled="${!stockMovement.pendingApproval}"
+                            disabledMessage="Request is not pending approval"
                     >
-                        <img src="${resource(dir: 'images/icons/silk', file: 'delete.png')}" />&nbsp;
-                        <warehouse:message code="default.button.delete.label" />
+                        <img src="${resource(dir: 'images/icons/silk', file: 'accept.png')}" />&nbsp;
```

Comment:
Is this the only place we use this icon on this page? If so accept.png (approved) and decline.png (rejected) should work perfectly. Otherwise, we might want to consider using a different pair (like tick.png, cross.png).